cmd/pkg-unpack/main.go

```diff
@@ -17,63 +17,71 @@ limitations under the License.
 package main
 
 import (
+	"bufio"
+	"bytes"
 	"context"
 	"fmt"
+	"io"
 	"os"
 
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructuredscheme"
-
 	"github.com/pkg/errors"
 	"github.com/spf13/afero"
-	"gopkg.in/yaml.v2"
-	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructuredscheme"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/util/yaml"
 
 	"github.com/crossplane/crossplane-runtime/pkg/parser"
 )
 
 func main() {
-	if err := Run(context.Background()); err != nil {
+	be := parser.NewFsBackend(afero.NewReadOnlyFs(afero.NewOsFs()), parser.FsDir("."), parser.FsFilters(parser.SkipNotYAML()))
+	src, err := be.Init(context.Background())
+	if err != nil {
+		// NOTE(negz): fmt.Print prints to stderr.
 		fmt.Print(err.Error())
 		os.Exit(1)
 	}
-}
+	defer src.Close()
 
-func Run(ctx context.Context) error {
-	uScheme := struct {
-		runtime.ObjectTyper
-		runtime.ObjectCreater
-	}{
-		ObjectTyper:   unstructuredscheme.NewUnstructuredObjectTyper(),
-		ObjectCreater: unstructuredscheme.NewUnstructuredCreator(),
-	}
-	p := parser.New(uScheme, uScheme)
```

Comment:
I think I caused confusion with my comments on the original PR, sorry about that folks. My intention was to indicate that I don't think we need to care about "linting" package contents in the unpack image. Doing so would mean taking a dependency on `crossplane/crossplane` and would cause the unpack image to need to (potentially) be updated more frequently when we are already planning on building linting into `crank` and invalid packages will be rejected by the package manager.

However, I do believe the parsing logic could be somewhat confusing here (everything is being loaded as `Meta` as @negz mentioned, https://github.com/crossplane/crossplane-runtime/blob/e1fc0493b2e88dc4b067634eed445cda345a9843/pkg/parser/parser.go#L104), but it seems like if we want to honor printing the `crossplane.yaml` first then we are going to inevitably have to parse with a scheme that recognizes meta separate from objects, which would violate the point above about not tying this to c/c.

On the other hand, the meta object is supposed to always be in `crossplane.yaml`, so we could just look for that and pull it out, which would not require a dependency on c/c. I believe this was part of the reason for having a unique name for that in the first place.